language/move-lang/src/parser/syntax.rs

```diff
@@ -1,4 +1,1412 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-include!(concat!(env!("OUT_DIR"), "/parser/syntax.rs"));
+use codespan::{ByteIndex, Span};
+use std::str::FromStr;
+
+use crate::parser::ast::*;
+use crate::parser::lexer::*;
+use crate::shared::*;
+
+// In the informal grammar comments in this file, Comma<T> is shorthand for:
+//      (<T> ",")* <T>?
+// Note that this allows an optional trailing comma.
+
+// FIXME: The following simplified version of ParseError copied from
+// lalrpop-util should be replaced.
+
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum ParseError<L, E> {
+    InvalidToken { location: L },
+    User { location: L, error: E },
+}
+
+fn make_loc(file: &'static str, start: usize, end: usize) -> Loc {
+    Loc::new(
+        file,
+        Span::new(ByteIndex(start as u32), ByteIndex(end as u32)),
+    )
+}
+
+fn spanned<T>(file: &'static str, start: usize, end: usize, value: T) -> Spanned<T> {
+    Spanned {
+        loc: make_loc(file, start, end),
+        value,
+    }
+}
+
+fn consume_token<'input>(
+    tokens: &mut Lexer<'input>,
+    tok: Tok,
+) -> Result<(), ParseError<usize, &'static str>> {
+    if tokens.peek() != tok {
+        return Err(ParseError::InvalidToken {
+            location: tokens.start_loc(),
```

Comment:
could we just add a really really simple improvement to the errors? 
```
let peeked = tokens.peek();
if tok != peeked {
  return Err(ParserError::InvalidToken { location: tokens.start_loc(), expected: tok, actual: peeked })
}
```

Reply:
I added back ParserError::UnrecognizedToken  with something like that.